### PR TITLE
feat: selector! macro

### LINKED ISSUE #643: Describe `selector!` macro
`selector!` macro computes sn_keccak and is very useful in contracts. For now, we can see this macro in one contract example, but without explanation.

https://book.cairo-lang.org/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.html?highlight=selector!#using-low-level-syscalls

Find the appropriate place and add informations to describe this macro.

### PR DESCRIPTION
**Pull request by JimmyFate**
Closes #643

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/653)

### CODE REVIEW

**Review by enitrat — COMMENTED** [review]
Hi, thanks for the PR! Can you also add info to the macros.md page?

**Comment by JimmyFate** [comment]
@enitrat Done :)
[mention selector! in macros.md](https://github.com/cairo-book/cairo-book/pull/653/commits/42e6e794c6865e320ada3fd69322b20c61b6c854)

**Comment by TAdev0** [comment]
@JimmyFate  i made a few comments. Can you modify what needs to be modified ?

**Comment by JimmyFate** [comment]
> @JimmyFate i made a few comments. Can you modify what needs to be modified ?

@TAdev0 do you mean comments on Reviewable? I don't see any

**Review by TAdev0 — CHANGES_REQUESTED** [review]

**Comment by TAdev0** [comment]
@JimmyFate  now it should be good. Sorry, i forgot to publish the review.

File: src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md [hunk 1]
```diff
@@ -89,6 +89,10 @@ Using these syscalls can be handy for customized error handling or to get more c
 {{#label syscalls}}
 <span class="caption">Listing {{#ref syscalls}}: A sample contract using syscalls</span>
 
-To use this syscall, we passed in the contract address, the selector of the function we want to call, and the call arguments.
+To use this syscall, we passed in the contract address, the function we want to call, and the call arguments.
```
[the remark this answers is not in the document]
Comment (JimmyFate) [reply src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md:92]: Indeed, but it seemed logical not to mention the specifics of how the function is passed, but rather say vaguely that it takes in the function to call, before the entry point selector is explained below.
Can add it back if you prefer though

File: src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md [hunk 2]
```diff
@@ -89,6 +89,10 @@ Using these syscalls can be handy for customized error handling or to get more c
...
 
 The call arguments must be provided as an array of `felt252`. To build this array, we serialize the expected function parameters into an `Array<felt252>` using the `Serde` trait, and then pass this array as calldata. At the end, we are returned a serialized value which we'll need to deserialize ourselves!
+
+In Starknet contract context, a **selector** is an identifier through which a function is callable in transactions or in other classes.
+
+Note that in `starknet::call_contract_syscall` we didn't specify the function name as a string (ByteArray), but rather used the `selector!` macro, which produces the selector by computing a `sn_keccak` hash on its input.
```
[the remark this answers is not in the document]
Comment (JimmyFate) [reply src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md]: [reword sentence](https://github.com/cairo-book/cairo-book/pull/653/commits/a2ade67cf5d19ca61287f32f3540b18ca750f24d)

File: src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md [hunk 3]
```diff
@@ -89,6 +89,10 @@ Using these syscalls can be handy for customized error handling or to get more c
 {{#label syscalls}}
 <span class="caption">Listing {{#ref syscalls}}: A sample contract using syscalls</span>
 
-To use this syscall, we passed in the contract address, the selector of the function we want to call, and the call arguments.
+To use this syscall, we passed in the contract address, the function we want to call, and the call arguments.
 
 The call arguments must be provided as an array of `felt252`. To build this array, we serialize the expected function parameters into an `Array<felt252>` using the `Serde` trait, and then pass this array as calldata. At the end, we are returned a serialized value which we'll need to deserialize ourselves!
+
+In Starknet contract context, a **selector** is an identifier through which a function is callable in transactions or in other classes.
```
[the remark this answers is not in the document]
Comment (JimmyFate) [reply src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md]: [remove unnecessary](https://github.com/cairo-book/cairo-book/pull/653/commits/6cda7b399164033e5ddb5f3eaeea0a28a774f9d2)

File: src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (TAdev0) [reply src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md:92]: @enitrat what do you think?

[the remark this answers is not in the document]
Comment (TAdev0) [reply src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md:92]: @JimmyFate  i think its better to add it back.  Moreover, the selector is introduced a few lines after that. You can add a small references to the entrypoint selector section if you want, or just add "(see next section)" or something like that.

**Review by TAdev0 — CHANGES_REQUESTED** [review]
@JimmyFate i think its better to add it back. Moreover, the selector is introduced a few lines after that. You can add a small reference to the entrypoint selector section if you want, or just add "(see next section)" or something like that.

File: src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md [hunk 4]
```diff
@@ -89,6 +89,12 @@ Using these syscalls can be handy for customized error handling or to get more c
...
 <span class="caption">Listing {{#ref syscalls}}: A sample contract using syscalls</span>
 
-To use this syscall, we passed in the contract address, the selector of the function we want to call, and the call arguments.
+To use this syscall, we passed in the contract address, the selector of the function we want to call (see next section), and the call arguments.
 
 The call arguments must be provided as an array of `felt252`. To build this array, we serialize the expected function parameters into an `Array<felt252>` using the `Serde` trait, and then pass this array as calldata. At the end, we are returned a serialized value which we'll need to deserialize ourselves!
+
+### Entry Point Selector
+
+In Starknet contract context, a **selector** is an identifier through which a function is callable.
```
Comment (enitrat) [inline src/ch14-02-contract-dispatchers-library-dispatchers-and-system-calls.md:98]:
```suggestion
In the context of a smart contract, a selector is a unique identifier for a specific entrypoint of a contract. When a transaction is sent to a contract, it includes the selector in the calldata to specify which function should be executed.

On Starknet, the selector is computed by applying the `sn_keccak` hash function to the string representation of the function name. If the function name is `transfer`, the selector can be computed with `selector!("transfer")`
```

**Review by enitrat — APPROVED** [review]
Great 📈

**Review by TAdev0 — APPROVED** [review]

**Pull request merged**